src/google.ts

```diff
@@ -0,0 +1,76 @@
+"user strict";
+
+import path = require("path");
+import process = require("process");
+import { google } from "googleapis";
+import { promises as fsPromises } from "fs";
+import { authenticate } from "@google-cloud/local-auth";
+import { OAuth2Client } from "google-auth-library";
+
+// Inspired by https://developers.google.com/sheets/api/quickstart/nodejs
+// Doc : https://developers.google.com/sheets/api/reference/rest
+
+// If modifying these scopes, delete file which is in GOOGLE_TOKEN_FILE_PATH.
+const SCOPES = ["https://www.googleapis.com/auth/spreadsheets.readonly"];
+// The file which is in GOOGLE_TOKEN_FILE_PATH, stores the user"s access and refresh tokens
+// and is created automatically when the authorization flow completes for the first time.
+const TOKEN_PATH = path.join(process.cwd(), process.env.GOOGLE_TOKEN_FILE_PATH as string);
+const CREDENTIALS_PATH = path.join(process.cwd(), process.env.GOOGLE_CREDENTIALS_PATH as string);
+
+async function loadSavedCredentialsIfExist(): Promise<any|null> {
```

Comment:
I put `any` type here cause i was facing of type issue between `JSONClient` and `OAuth2Client` line 42 when i tried:
```
async function authorize(): Promise<OAuth2Client | JSONClient> {
```

Reply:
A solution will be:
```
async function loadSavedCredentialsIfExist(): Promise<JSONClient|OAuth2Client|null> {...}
async function authorize(): Promise<OAuth2Client|JSONClient> {...}
async function listRestaurants(auth: OAuth2Client|JSONClient): Promise<void> {...}
```
But the loadSavedCredentialsIfExist function never return OAuth2Client type.

Reply:
do we need to store the credentials ? 
if not we could delete this method and we would not need to bother with this type definition warning

Reply:
Yes, because without the file, the authentication to the google api requires a manual action with a web browser.